week10/MovieViewer/app/src/main/java/com/wmw/movieviewer/repository/MoviesRepositoryImpl.kt

```diff
@@ -0,0 +1,45 @@
+package com.wmw.movieviewer.repository
+
+import androidx.lifecycle.LiveData
+import com.wmw.movieviewer.model.Movie
+import com.wmw.movieviewer.model.MovieDao
+import com.wmw.movieviewer.model.Success
+import com.wmw.movieviewer.model.response.MovieResponse
+import com.wmw.movieviewer.networking.MoviesApi
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+
+open class MoviesRepositoryImpl : MoviesRepository, KoinComponent {
```

Comment:
You don't need the KoinComponent, and you already have it in the MoviesRepository interface...

The point of dependency injection is to pass stuff through the constructor, as we've been saying for quite a while now, not getting dependencies as the call site.